### PR TITLE
OFMCC-850 - Application start page

### PR DESCRIPTION
**Pull request by vietle-cgi**
Add Applications table to ApplicationsView

### CODE REVIEW

File: frontend/src/services/applicationService.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import { ApiRoutes } from '@/utils/constants'
+import ApiService from '@/common/apiService'
+
+export default {
+  sortApplications(applications) {
+    applications?.sort((app1, app2) => {
+      const latestActivityDateOrder = new Date(app2.latestActivity) - new Date(app1.latestActivity)
+      return latestActivityDateOrder
```
Comment (weskubo-cgi) [inline frontend/src/services/applicationService.js:8]: This is personal preference but I won't typically define a variable if it's only used once. At that point your code is one line and you can use an implicit return.

**Review by weskubo-cgi — CHANGES_REQUESTED** [review]

File: backend/src/components/applications.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
+'use strict'
+const { getOperation } = require('./utils')
+const { MappableObjectForFront } = require('../util/mapping/MappableObject')
+const { ApplicationMappings } = require('../util/mapping/Mappings')
+const HttpStatus = require('http-status-codes')
+const log = require('./logger')
+
+function mapApplicationObjectForFront(data) {
+  let application = new MappableObjectForFront(data, ApplicationMappings).toJSON()
```
[the remark this answers is not in the document]
Comment (vietle-cgi) [reply backend/src/components/applications.js]: it is updated

File: backend/src/components/applications.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  const ministryLastUpdated = new Date(application?.ministryLastUpdated)
+  const providerLastUpdated = new Date(application?.providerLastUpdated)
+  application.latestActivity = ministryLastUpdated > providerLastUpdated ? ministryLastUpdated : providerLastUpdated
+  return application
+}
+
+async function getApplicationsByFacilityId(req, res) {
+  try {
+    log.debug('getApplicationsByFacilityId: ', req.params.facilityId)
+    let applications = []
```
[the remark this answers is not in the document]
Comment (vietle-cgi) [reply backend/src/components/applications.js]: it is updated

File: frontend/src/views/ApplicationsView.vue [hunk 4]
```diff
@@ -1,7 +1,102 @@
 <template>
-  <h1>Applications</h1>
+  <v-container fluid>
+    <h3>Applications</h3>
```
[the remark this answers is not in the document]
Comment (vietle-cgi) [reply frontend/src/views/ApplicationsView.vue]: it is updated

File: frontend/src/services/applicationService.js (hunk 1 above)
Comment (vietle-cgi) [reply frontend/src/services/applicationService.js]: it is updated

File: frontend/src/views/ApplicationsView.vue [hunk 5]
```diff
@@ -1,7 +1,102 @@
 <template>
```
[the remark this answers is not in the document]
Comment (weskubo-cgi) [reply frontend/src/views/ApplicationsView.vue:1]: To be completed in a separate branch along with other pending unit tests.

File: frontend/src/views/ApplicationsView.vue [hunk 6]
```diff
@@ -1,7 +1,102 @@
...
+        </template>
+        <template #item.latestActivity="{ item }">
+          {{ format.formatDate(item.latestActivity) }}
+        </template>
+        <template #item.actionButtons="{ item }">
+          <v-btn v-if="isApplicationDownloadable(item)" variant="text" @click="false">
+            <v-icon icon="fa:fa-regular fa-file-pdf"></v-icon>
+          </v-btn>
+          <v-btn v-if="isApplicationDeletable(item)" variant="text" @click="false">
+            <v-icon icon="fa:fa-regular fa-trash-can"></v-icon>
```
[the remark this answers is not in the document]
Comment (weskubo-cgi) [reply frontend/src/views/ApplicationsView.vue:23]: FA trash icon was chosen for style/size compatibility with pdf icon. Fine as is.

**Review by weskubo-cgi — APPROVED** [review]

**Pull request merged**